src/instance/mod.rs

```diff
@@ -54,21 +66,22 @@ impl ObjectInstance {
 
     /// Create a new instance from raw bytes instead of a vector
     pub fn from_bytes(
-        ty: Option<Ty>,
+        ty: Option<TypeDec>,
         size: usize,
         data: &[u8],
-        fields: Option<Vec<(Name, Size)>>,
+        fields: Option<Vec<(Name, ObjectInstance)>>,
     ) -> ObjectInstance {
         ObjectInstance::new(ty, size, data.to_vec(), fields)
     }
 
     /// Get a reference to the type of the instance
-    pub fn ty(&self) -> Option<&Ty> {
-        self.ty.as_ref()
+    // FIXME: Remove clone
+    pub fn ty(&self) -> Option<TypeDec> {
+        self.ty.clone()
     }
```

Comment:
What makes it impossible to use a ref ?

Reply:
Good catch, thanks. Actually nothing. This was from the first WIP on this, as I was battling with some other stuff